### PR TITLE
fix(zetacore): bring back reserved fields as deprecated

### PR DESCRIPTION
**Pull request by swift1337**
Cosmos SDK's proto decoder explicitly fails to decode fields marked as "reserved." This PR reinstates legacy fields as deprecated. ~~Deprecated fields should also be excluded from JSON and YAML encoders~~

Closes https://github.com/zeta-chain/node/issues/3280

## Summary by CodeRabbit

- **New Features**
  - Added a new boolean property `proved` to the OpenAPI specifications.
  - Enhanced tracking capabilities for cross-chain transactions with new fields in the `MsgAddOutboundTracker` and `MsgAddInboundTracker` messages, including proof and indexing information.

- **Bug Fixes**
  - Deprecated previous fields in the `TxHash`, `MsgAddOutboundTracker`, and `MsgAddInboundTracker` messages to improve clarity and functionality.

- **Documentation**
  - Updated documentation to reflect changes in message definitions and new properties.

### CODE REVIEW

File: x/crosschain/types/tx.pb.go [hunk 1]
```diff
@@ -210,10 +210,13 @@ func (m *MsgUpdateTssAddressResponse) XXX_DiscardUnknown() {
...
 type MsgAddInboundTracker struct {
-	Creator  string        `protobuf:"bytes,1,opt,name=creator,proto3" json:"creator,omitempty"`
-	ChainId  int64         `protobuf:"varint,2,opt,name=chain_id,json=chainId,proto3" json:"chain_id,omitempty"`
-	TxHash   string        `protobuf:"bytes,3,opt,name=tx_hash,json=txHash,proto3" json:"tx_hash,omitempty"`
-	CoinType coin.CoinType `protobuf:"varint,4,opt,name=coin_type,json=coinType,proto3,enum=zetachain.zetacore.pkg.coin.CoinType" json:"coin_type,omitempty"`
+	Creator   string        `protobuf:"bytes,1,opt,name=creator,proto3" json:"creator,omitempty"`
+	ChainId   int64         `protobuf:"varint,2,opt,name=chain_id,json=chainId,proto3" json:"chain_id,omitempty"`
+	TxHash    string        `protobuf:"bytes,3,opt,name=tx_hash,json=txHash,proto3" json:"tx_hash,omitempty"`
+	CoinType  coin.CoinType `protobuf:"varint,4,opt,name=coin_type,json=coinType,proto3,enum=zetachain.zetacore.pkg.coin.CoinType" json:"coin_type,omitempty"`
+	Proof     *proofs.Proof `protobuf:"bytes,5,opt,name=proof,proto3" json:"-" yaml:"-"`                     // Deprecated: Do not use.
```
Comment (swift1337) [inline x/crosschain/types/tx.pb.go:217]: Note `json:"-" yaml:"-"`

File: proto/zetachain/zetacore/crosschain/outbound_tracker.proto [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+import "gogoproto/gogo.proto";
 option go_package = "github.com/zeta-chain/node/x/crosschain/types";
 
 message TxHash {
   string tx_hash = 1;
   string tx_signer = 2;
 
-  // used to be `bool proven` (for block header verification)
-  reserved 3;
+  bool proved = 3 [
```
Comment (kingpinXD) [inline proto/zetachain/zetacore/crosschain/outbound_tracker.proto:11]: Should this be `proven` going by the older comment

**Comment by swift1337** [comment]
@kingpinXD @fadeev @CharlieMc0 

```sh
zetacored q tx 6C86B759289ACB1BE2EEC35619B7F9227EC4AF4AC235BC70F366C1DCDF76539A --node=https://zetachain-mainnet.g.allthatnode.com:443/archive/tendermint -o json | jq
```

```json
{
  "height": "6057922",
  "txhash": "6C86B759289ACB1BE2EEC35619B7F9227EC4AF4AC235BC70F366C1DCDF76539A",
  "codespace": "",
  "code": 0,
  "data": "122B0A252F636F736D6F732E617574687A2E763162657461312E4D736745786563526573706F6E736512020A00",
  "raw_log": "[{\"msg_index\":0,\"events\":[{\"type\":\"message\",\"attributes\":[{\"key\":\"action\",\"value\":\"/cosmos.authz.v1beta1.MsgExec\"},{\"key\":\"sender\",\"value\":\"zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k\"},{\"key\":\"module\",\"value\":\"authz\"}]}]}]",
  "logs": [
    {
      "msg_index": 0,
      "log": "",
      "events": [
        {
          "type": "message",
          "attributes": [
            {
              "key": "action",
              "value": "/cosmos.authz.v1beta1.MsgExec"
            },
            {
              "key": "sender",
              "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k"
            },
            {
              "key": "module",
              "value": "authz"
            }
          ]
        }
      ]
    }
  ],
  "info": "",
  "gas_wanted": "200000",
  "gas_used": "104673",
  "tx": {
    "@type": "/cosmos.tx.v1beta1.Tx",
    "body": {
      "messages": [
        {
          "@type": "/cosmos.authz.v1beta1.MsgExec",
          "grantee": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "msgs": [
            {
              "@type": "/zetachain.zetacore.crosschain.MsgAddOutboundTracker",
              "creator": "zeta19jr7nl82lrktge35f52x9g5y5prmvchmk40zhg",
              "chain_id": "1",
              "nonce": "46275",
              "tx_hash": "0x878345cdd31740ce8c14c0638b66873911be39889afe654e3301dc47dcf1717b",
              "proof": null,
              "block_hash": "",
              "tx_index": "-1"
            }
          ]
        }
      ],
      "memo": "",
      "timeout_height": "0",
      "extension_options": [],
      "non_critical_extension_options": []
    },
    "auth_info": {
      "signer_infos": [
        {
          "public_key": {
            "@type": "/ethermint.crypto.v1.ethsecp256k1.PubKey",
            "key": "A7JcN5KwyoI7U+WkuklFZZE//e9GiWow5IuFRaMoTUlr"
          },
          "mode_info": {
            "single": {
              "mode": "SIGN_MODE_DIRECT"
            }
          },
          "sequence": "255890"
        }
      ],
      "fee": {
        "amount": [
          {
            "denom": "azeta",
            "amount": "30000000000000"
          }
        ],
        "gas_limit": "200000",
        "payer": "",
        "granter": ""
      },
      "tip": null
    },
    "signatures": [
      "qlf6++n0CyhLTULG358j1JEm7WCsRf0FRevDU5hwhEcpAZ+Z9occNa1ASGrBNVJvla30fqRc73LbjULTB43IggE="
    ]
  },
  "timestamp": "2024-12-07T14:56:47Z",
  "events": [
    {
      "type": "coin_spent",
      "attributes": [
        {
          "key": "spender",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "index": true
        },
        {
          "key": "amount",
          "value": "30000000000000azeta",
          "index": true
        }
      ]
    },
    {
      "type": "coin_received",
      "attributes": [
        {
          "key": "receiver",
          "value": "zeta17xpfvakm2amg962yls6f84z3kell8c5lxad43d",
          "index": true
        },
        {
          "key": "amount",
          "value": "30000000000000azeta",
          "index": true
        }
      ]
    },
    {
      "type": "transfer",
      "attributes": [
        {
          "key": "recipient",
          "value": "zeta17xpfvakm2amg962yls6f84z3kell8c5lxad43d",
          "index": true
        },
        {
          "key": "sender",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "index": true
        },
        {
          "key": "amount",
          "value": "30000000000000azeta",
          "index": true
        }
      ]
    },
    {
      "type": "message",
      "attributes": [
        {
          "key": "sender",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "index": true
        }
      ]
    },
    {
      "type": "tx",
      "attributes": [
        {
          "key": "fee",
          "value": "30000000000000azeta",
          "index": true
        },
        {
          "key": "fee_payer",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "index": true
        }
      ]
    },
    {
      "type": "tx",
      "attributes": [
        {
          "key": "acc_seq",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k/255890",
          "index": true
        }
      ]
    },
    {
      "type": "tx",
      "attributes": [
        {
          "key": "signature",
          "value": "qlf6++n0CyhLTULG358j1JEm7WCsRf0FRevDU5hwhEcpAZ+Z9occNa1ASGrBNVJvla30fqRc73LbjULTB43IggE=",
          "index": true
        }
      ]
    },
    {
      "type": "message",
      "attributes": [
        {
          "key": "action",
          "value": "/cosmos.authz.v1beta1.MsgExec",
          "index": true
        },
        {
          "key": "sender",
          "value": "zeta15l7fehe8azkmj6979ln7e74dkvl0ypledqje7k",
          "index": true
        },
        {
          "key": "module",
          "value": "authz",
          "index": true
        }
      ]
    }
  ]
}
```

**Comment by kingpinXD** [comment]
Thanks , verfied that the tx can be quired with the new binary , and also that the tx cannot be queried with the older one .

**Review by lumtis — APPROVED** [review]

**Pull request merged**